### PR TITLE
add Vagrantfile and pin requirements

### PR DESCRIPTION
**Pull request by erickyellott**
Adds a Vagrantfile for local dev.

Usage:

```
$ vagrant up
$ vagrant ssh
...
vagrant@ubuntu-bionic:/vagrant/proj$ ./manage.py runserver 0:8000
Performing system checks...

System check identified no issues (0 silenced).
October 04, 2018 - 12:16:07
Django version 2.1, using settings 'proj.settings'
Starting development server at http://0:8000/
Quit the server with CONTROL-C.
[04/Oct/2018 12:16:09] "GET / HTTP/1.1" 200 1466
[04/Oct/2018 12:16:09] "GET /static/css/base.css HTTP/1.1" 200 1916
[04/Oct/2018 12:16:09] "GET /static/js/jquery-3.3.1.min.js HTTP/1.1" 200 86927
[04/Oct/2018 12:16:09] "GET /static/js/intercooler-1.2.1.min.js HTTP/1.1" 200 31367
```

screen shot 2018-10-04 at 7 21 48 am

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,55 @@
 Django==2.1
 feedparser==5.2.1
 pytz==2018.5
+## The following requirements were added by pip freeze:
```
Comment (svsaraf) [inline requirements.txt]: wait where did these come from?

File: proj/proj/settings.py [hunk 2]
```diff
@@ -20,12 +20,12 @@
 # See https://docs.djangoproject.com/en/2.1/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'qzuu+0en!psjk88led)^@_vrf4%l9u4r)j-^+%h4esw-$#_e24'
+SECRET_KEY = 'qzuu+0en!psjk88led)^@_vrf4%l9u4r)j-^+%h4esw-$#_e24'  # TODO: rotate this key
```
Comment (svsaraf) [inline proj/proj/settings.py:23]: ack, yeah i need to do this in prod

File: requirements.txt (hunk 1 above)
Comment (svsaraf) [reply requirements.txt]: When I run pip freeze I only get the three requirements above. Are you in a virtualenv? Maybe these are system wide pip installs?

Comment (erickyellott) [reply requirements.txt]: oh christ. these are from the ubuntu/bionic64 base image.
didn't think the system packages would be so polluted, my b

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# -*- mode: ruby -*-
```
Comment (svsaraf) [inline Vagrantfile:1]: this is cool! i've never used vagrant before.

File: requirements.txt (hunk 1 above)
Comment (svsaraf) [reply requirements.txt]: Haha all good, will merge once adjusted

**Review by svsaraf — APPROVED** [review]

**Comment by svsaraf** [comment]
Thanks!! You're my hero!

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  config.vm.provision "shell", inline: <<-SHELL
+    set -e
+
+    ln -sf /usr/bin/python3 /usr/bin/python
+    apt-get update && apt-get install -y python3-pip
+    pip3 install -U pip
+    pip install virtualenv
+    virtualenv /vagrant/.venv
+    /vagrant/.venv/bin/pip3 install -r /vagrant/requirements.txt
+    chown -R vagrant:vagrant /home/vagrant/.venv
```
Comment (erickyellott) [inline Vagrantfile:37]: ah, sorry, there's one last bug here

File: Vagrantfile (hunk 3 above)
Comment (erickyellott) [reply Vagrantfile:1]: everyone's on the docker train atm, but vagrant is still my goto for small projects.

**Pull request merged**